### PR TITLE
Merge rc/1.7.0 into staking v4 3 jan 2024

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Merge rc/1.7.0 into staking v4 3 jan 2024

### CODE REVIEW

File: config/systemSmartContractsConfig.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ type ESDTSystemSCConfig struct {
 }
 
 // GovernanceSystemSCConfigV1 holds the initial set of values that were used to initialise the
-//  governance system smart contract at genesis time
+//
+//	governance system smart contract at genesis time
```
Comment (sstanculeanu) [inline config/systemSmartContractsConfig.go:39]: wrong indentation

File: config/systemSmartContractsConfig.go [hunk 2]
```diff
@@ -45,7 +46,8 @@ type GovernanceSystemSCConfigV1 struct {
 }
 
 // GovernanceSystemSCConfigActive defines the set of configuration values used by the governance
-//  system smart contract once it activates
+//
```
Comment (sstanculeanu) [inline config/systemSmartContractsConfig.go:49]: wrong indentation

File: epochStart/metachain/systemSCs.go [hunk 3]
```diff
@@ -69,7 +65,8 @@ func NewSystemSCProcessor(args ArgsNewEpochStartSystemSCProcessing) (*systemSCPr
 	if check.IfNil(args.EnableEpochsHandler) {
 		return nil, epochStart.ErrNilEnableEpochsHandler
 	}
-	err := core.CheckHandlerCompatibility(args.EnableEpochsHandler, []core.EnableEpochFlag{
+
+	err = core.CheckHandlerCompatibility(args.EnableEpochsHandler, []core.EnableEpochFlag{
```
Comment (sstanculeanu) [inline epochStart/metachain/systemSCs.go:69]: should also add here: StakingV4Step1Flag, StakingV4Step2Flag

File: epochStart/metachain/systemSCs_test.go [hunk 4]
```diff
@@ -2091,7 +2089,11 @@ func TestSystemSCProcessor_LegacyEpochConfirmedCorrectMaxNumNodesAfterNodeRestar
 			nodesConfigEpoch6,
 		})
 	args.MaxNodesChangeConfigProvider = nodesConfigProvider
-	args.EnableEpochsHandler = &testscommon.EnableEpochsHandlerStub{IsStakingV2FlagEnabledField: true}
+	args.EnableEpochsHandler = &enableEpochsHandlerMock.EnableEpochsHandlerStub{
```
Comment (sstanculeanu) [inline epochStart/metachain/systemSCs_test.go:2092]: could have been used enableEpochsHandlerMock.NewEnableEpochsHandler(common.StakingV2Flag)

File: factory/api/apiResolverFactory.go [hunk 5]
```diff
@@ -470,12 +470,12 @@ func createMetaVmContainerFactory(args *scQueryElementArgs, argsHook hooks.ArgBl
 		Marshalizer:         args.coreComponents.InternalMarshalizer(),
 		SystemSCConfig:      args.systemSCConfig,
 		ValidatorAccountsDB: args.stateComponents.PeerAccounts(),
-		UserAccountsDB:      args.stateComponents.AccountsAdapterAPI(),	ChanceComputer:      args.coreComponents.Rater(),
-			ShardCoordinator:    args.processComponents.ShardCoordinator(),
-			EnableEpochsHandler: args.coreComponents.EnableEpochsHandler(),
-			NodesCoordinator:    args.processComponents.NodesCoordinator(),
-		}
-		vmFactory, err := metachain.NewVMContainerFactory(argsNewVmFactory)
+		UserAccountsDB:      args.stateComponents.AccountsAdapterAPI(), ChanceComputer: args.coreComponents.Rater(),
```
Comment (sstanculeanu) [inline factory/api/apiResolverFactory.go:473]: 2 args on the same line

File: integrationTests/vm/txsFee/scCalls_test.go [hunk 6]
```diff
@@ -59,10 +59,10 @@ func prepareTestContextForEpoch836(tb testing.TB) (*vm.VMTestContext, []byte) {
 
 	testContext, err := vm.CreatePreparedTxProcessorWithVMsWithShardCoordinatorDBAndGasAndRoundConfig(
 		config.EnableEpochs{
-			GovernanceEnableEpoch:                           unreachableEpoch,
+			GovernanceEnableEpoch:                   unreachableEpoch,
 			SetSenderInEeiOutputTransferEnableEpoch: unreachableEpoch,
 			RefactorPeersMiniBlocksEnableEpoch:      unreachableEpoch,
-			MaxBlockchainHookCountersEnableEpoch:    unreachableEpoch,DynamicGasCostForDataTrieStorageLoadEnableEpoch: unreachableEpoch,
+			MaxBlockchainHookCountersEnableEpoch:    unreachableEpoch, DynamicGasCostForDataTrieStorageLoadEnableEpoch: unreachableEpoch,
```
Comment (sstanculeanu) [inline integrationTests/vm/txsFee/scCalls_test.go:65]: 2 args on the same line

File: state/interface.go [hunk 7]
```diff
@@ -24,7 +24,8 @@ type Updater interface {
 }
 
 // PeerAccountHandler models a peer state account, which can journalize a normal account's data
-//  with some extra features like signing statistics or rating information
+//
+//	with some extra features like signing statistics or rating information
```
Comment (sstanculeanu) [inline state/interface.go:28]: wrong indentation

File: config/systemSmartContractsConfig.go (hunk 1 above)
Comment (mariusmihaic) [reply config/systemSmartContractsConfig.go:39]: Right, good catch

File: config/systemSmartContractsConfig.go (hunk 2 above)
Comment (mariusmihaic) [reply config/systemSmartContractsConfig.go:49]: Right, good catch

File: epochStart/metachain/systemSCs.go (hunk 3 above)
Comment (mariusmihaic) [reply epochStart/metachain/systemSCs.go:69]: Added them

File: epochStart/metachain/systemSCs_test.go (hunk 4 above)
Comment (mariusmihaic) [reply epochStart/metachain/systemSCs_test.go:2092]: RIght, refactored as suggested

File: factory/api/apiResolverFactory.go (hunk 5 above)
Comment (mariusmihaic) [reply factory/api/apiResolverFactory.go:473]: Right, good catch

File: integrationTests/vm/txsFee/scCalls_test.go (hunk 6 above)
Comment (mariusmihaic) [reply integrationTests/vm/txsFee/scCalls_test.go:65]: Right, good catch

File: state/interface.go (hunk 7 above)
Comment (mariusmihaic) [reply state/interface.go:28]: Right, good catch

**Review by sstanculeanu — APPROVED** [review]

**Review by sasurobert — COMMENTED** [review]
only sorins comments.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**